test/functional/feature_avoidreuse.py

```diff
@@ -0,0 +1,92 @@
+#!/usr/bin/env python3
+# Copyright (c) 2017 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+"""Test the -avoidreuse config option."""
+
+from test_framework.test_framework import BitcoinTestFramework
+from test_framework.util import assert_raises_rpc_error
+from decimal import Decimal
+
+class AvoidReuseTest(BitcoinTestFramework):
+
+    def set_test_params(self):
+        self.setup_clean_chain = False
+        self.num_nodes = 2
+        self.extra_args = [[], ['-avoidreuse=1']]
+
+    def reset_balance(self, node, discardaddr):
+        '''
+        Throw away all owned coins by the node so it gets a balance of 0.
+        '''
+        balance = node.getbalance()
+        if balance > 0.5:
+            tosend = '%.5f' % (balance - Decimal(0.01))
```

Comment:
Commit "test: Add test for avoidreuse feature"

Use `subtractfeefromamount=true` below instead?